packages/zwave-js/src/lib/driver/FirmwareUpdateService.ts

```diff
@@ -0,0 +1,98 @@
+import {
+	extractFirmware,
+	Firmware,
+	guessFirmwareFileFormat,
+	ZWaveError,
+	ZWaveErrorCodes,
+} from "@zwave-js/core";
+import { formatId } from "@zwave-js/shared";
+import axios from "axios";
+import crypto from "crypto";
+import type { FirmwareUpdateFileInfo, FirmwareUpdateInfo } from "./_Types";
+
+// const serviceURL = "https://firmware.zwave-js.io";
+const serviceURL = "http://localhost:3000";
+
+/**
+ * Retrieves the available firmware updates for the node with the given fingerprint.
+ * Returns the service response or `undefined` in case of an error.
+ */
+export async function getAvailableFirmwareUpdates(
+	manufacturerId: number,
+	productType: number,
+	productId: number,
+	firmwareVersion: string,
+): Promise<FirmwareUpdateInfo[]> {
+	const response: FirmwareUpdateInfo[] = (
+		await axios({
+			method: "post",
+			url: `${serviceURL}/api/v1/updates`,
+			data: {
+				manufacturerId: formatId(manufacturerId),
+				productType: formatId(productType),
+				productId: formatId(productId),
+				firmwareVersion,
+			},
+		})
+	).data;
+
+	return response;
+}
+
+export async function downloadFirmwareUpdate(
+	file: FirmwareUpdateFileInfo,
+): Promise<Firmware> {
+	const [hashAlgorithm, expectedHash] = file.integrity.split(":", 2);
+
+	if (hashAlgorithm !== "sha256") {
+		throw new ZWaveError(
+			`Unsupported hash algorithm ${hashAlgorithm} for integrity check`,
+			ZWaveErrorCodes.Argument_Invalid,
+		);
+	}
+	// TODO: Make request abort-able (requires AbortController, Node 14.17+ / Node 16)
+
+	// Download the firmware file
+	const downloadResponse = await axios.get<Buffer>(file.url, {
+		timeout: 60000,
+		maxContentLength: 10 * 1024 * 1024, // 10MB should be enough for any conceivable Z-Wave chip
```

Comment:
I personally would place both this in a constant instead like `DOWNLOAD_TIMEOUT` and `MAX_UPDATE_SIZE`